### PR TITLE
chore(samples): Snippet Generating Script: part 2 - moving the snippets to new format

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-compute/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

BEGIN_COMMIT_OVERRIDE
docs(samples): additional samples for the Compute API
END_COMMIT_OVERRIDE

### CODE REVIEW

File: samples/recipes/usage_report/usage_reports.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# flake8: noqa
+"""
+A sample script showing how to handle default values when communicating
+with the Compute Engine API.
```
Comment (parthea) [inline samples/recipes/usage_report/usage_reports.py:17]: Should the comment also include a note about usage reports?

File: samples/snippets/firewall/create.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# This file is automatically generated. Please do not modify it directly.
+# Find the relevant recipe file in the samples/recipes or samples/ingredients
+# directory and apply your changes there.
+
+
+# [START compute_firewall_create]
+from google.cloud import compute_v1
+
+
+def create_firewall_rule(
```
Comment (parthea) [inline samples/snippets/firewall/create.py:26]: The tag `compute_firewall_create` doesn't match `create_firewall_rule` . Is that expected?

```suggestion
def compute_firewall_create(
```

File: samples/snippets/firewall/create.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    """
+    Creates a simple firewall rule allowing for incoming HTTP and HTTPS access from the entire Internet.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        firewall_rule_name: name of the rule that is created.
+        network: name of the network the rule will be applied to. Available name formats:
+            * https://www.googleapis.com/compute/v1/projects/{project_id}/global/networks/{network}
+            * projects/{project_id}/global/networks/{network}
+            * global/networks/{network}
```
Comment (parthea) [inline samples/snippets/firewall/create.py:38]: Add return value in docstring

File: samples/snippets/firewall/create.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+# [START compute_firewall_create]
+from google.cloud import compute_v1
+
+
+def create_firewall_rule(
+    project_id: str, firewall_rule_name: str, network: str = "global/networks/default"
+) -> compute_v1.Firewall:
+    """
+    Creates a simple firewall rule allowing for incoming HTTP and HTTPS access from the entire Internet.
```
Comment (parthea) [inline samples/snippets/firewall/create.py:30]: Should we add a note about the return value?

File: samples/snippets/firewall/create.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    allowed_ports.ports = ["80", "443"]
+
+    firewall_rule.allowed = [allowed_ports]
+    firewall_rule.source_ranges = ["0.0.0.0/0"]
+    firewall_rule.network = network
+    firewall_rule.description = "Allowing TCP traffic on port 80 and 443 from Internet."
+
+    firewall_rule.target_tags = ["web"]
+
+    # Note that the default value of priority for the firewall API is 1000.
```
Comment (parthea) [inline samples/snippets/firewall/create.py:55]: Should we add a TODO(Developer) prefix here if we want developers to uncomment the code?

File: samples/snippets/firewall/delete.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+# directory and apply your changes there.
+
+
+# [START compute_firewall_delete]
+from google.cloud import compute_v1
+
+
+def delete_firewall_rule(project_id: str, firewall_rule_name: str):
+    """
+    Deleted a firewall rule from the project.
```
Comment (parthea) [inline samples/snippets/firewall/delete.py:28]:
```suggestion
    Deletes a firewall rule from the project.
```

File: samples/snippets/firewall/delete.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+# This file is automatically generated. Please do not modify it directly.
+# Find the relevant recipe file in the samples/recipes or samples/ingredients
+# directory and apply your changes there.
+
+
+# [START compute_firewall_delete]
+from google.cloud import compute_v1
+
+
+def delete_firewall_rule(project_id: str, firewall_rule_name: str):
```
Comment (parthea) [inline samples/snippets/firewall/delete.py:26]:
```suggestion
def delete_firewall_rule(project_id: str, firewall_rule_name: str) -> None:
```

File: samples/snippets/firewall/delete.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    firewall_client = compute_v1.FirewallsClient()
+    operation = firewall_client.delete_unary(
+        project=project_id, firewall=firewall_rule_name
+    )
+
+    operation_client = compute_v1.GlobalOperationsClient()
+    operation_client.wait(project=project_id, operation=operation.name)
+    return
+
+
```
Comment (parthea) [inline samples/snippets/firewall/delete.py:43]:
```suggestion
```

File: samples/snippets/firewall/patch.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+# This file is automatically generated. Please do not modify it directly.
+# Find the relevant recipe file in the samples/recipes or samples/ingredients
+# directory and apply your changes there.
+
+
+# [START compute_firewall_patch]
+from google.cloud import compute_v1
+
+
+def patch_firewall_priority(project_id: str, firewall_rule_name: str, priority: int):
```
Comment (parthea) [inline samples/snippets/firewall/patch.py:26]:
```suggestion
def patch_firewall_priority(project_id: str, firewall_rule_name: str, priority: int) -> None:
```

File: samples/snippets/instances/create.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+# [START compute_instances_create]
+import re
+import sys
+from typing import List
+
+from google.cloud import compute_v1
+
+
+def get_image_from_family(project: str, family: str) -> compute_v1.Image:
```
Comment (parthea) [inline samples/snippets/instances/create.py:30]: Missing docstring

File: samples/snippets/instances/custom_machine_types/create_with_helper.py [hunk 11]
```diff
@@ -206,16 +211,58 @@ def from_str(cls, machine_type: str):
...
+
+def disk_from_image(
+    disk_type: str, disk_size_gb: int, boot: bool, source_image: str
+) -> compute_v1.AttachedDisk:
+    """
+    Create an AttachedDisk object to be used in VM instance creation. Uses an image as the
+    source for the new disk.
+
+    Args:
+         disk_type: the type of disk you want to create. This value uses the following format:
```
Comment (parthea) [inline samples/snippets/instances/custom_machine_types/create_with_helper.py:229]:
```suggestion
        disk_type: the type of disk you want to create. This value uses the following format:
```

File: samples/snippets/instances/custom_machine_types/create_with_helper.py [hunk 12]
```diff
@@ -226,49 +273,72 @@ def create_instance(
         machine_type: machine type of the VM being created. This value uses the
             following format: "zones/{zone}/machineTypes/{type_name}".
             For example: "zones/europe-west3-c/machineTypes/f1-micro"
-            OR
-            It can be a CustomMachineType object, describing a custom type
-            you want to use.
-
-    Return:
+        disks: a list of compute_v1.AttachedDisk objects describing the disks
```
Comment (parthea) [inline samples/snippets/instances/custom_machine_types/create_with_helper.py:276]: move `disks` above `machine_type`

**Review by parthea — COMMENTED** [review]
added minor nits, otherwise LGTM

File: samples/ingredients/instances/create_start_instance/create_from_custom_image.py [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+) -> compute_v1.Instance:
+    """
+    Create a new VM instance with custom image used as its boot disk.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        instance_name: name of the new virtual machine (VM) instance.
+        custom_image_link: link to the custom image you want to use in the form of:
+            "projects/{project_name}/global/images/{image_name}"
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/ingredients/instances/create_start_instance/create_from_custom_image.py:44]: I'd rather is stay as project_name, so it is not confused with the argument `project_id`. The project in this string may be referencing an image from a *different* project than the one specified by `project_id`. `project_id` tells the system in which project to create a new VM, while the `custom_image_link` points at the image that is to be used.

File: samples/recipes/operations/operation_check.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# flake8: noqa
+
+# <REGION compute_instances_operation_check>
+# <IMPORTS/>
+
+# <INGREDIENT wait_for_operation/>
+# </REGION compute_instances_operation_check>
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/recipes/operations/operation_check.py:33]: This is not a test. This file is used to produce the `samples/snippets/operations/operation_check.py` file using in this case one ingredient `wait_for_operation` in a region `compute_instances_operation_check`.

Same with all the other files in `recipes` folder, they are the instructions how to make the final code snippets from ingredients.

File: samples/snippets/firewall/create.py (hunk 2 above)
Comment (m-strzelczyk) [reply samples/snippets/firewall/create.py:26]: Yes, that is intentional. Tag needs to start with a product, in this case `compute`, while the method names usually start with a verb. The region tags don't have to match the function names.

**Review by parthea — APPROVED** [review]
LGTM

**Pull request merged**